### PR TITLE
[MU4] General menu. Format

### PR DESCRIPTION
**Pull request by Eism**
Implemented:
Increase Layout Stretch
Decrease Layout Stretch
Reset Layout Stretch
Reset Text Style Overrides
Reset Beams
Reset Shapes and Positions

### CODE REVIEW

File: src/appshell/view/appmenumodel.cpp [hunk 1]
```diff
@@ -250,23 +260,23 @@ MenuItem AppMenuModel::addItem()
 MenuItem AppMenuModel::formatItem()
```
Comment (RomanPudashkin) [inline src/appshell/view/appmenumodel.cpp:260]: Can you just disable the format menu if there is no opened score?

File: src/notation/internal/notationactioncontroller.cpp [hunk 2]
```diff
@@ -892,6 +901,70 @@ void NotationActionController::addAnchoredNoteLine()
...
+    if (!interaction) {
+        return;
+    }
+
+    if (interaction->selection()->isNone() || interaction->selection()->isRange()) {
+        interaction->resetBeamMode();
+    }
+}
+
+void NotationActionController::resetShapesAndPosition()
```
Comment (RomanPudashkin) [inline src/notation/internal/notationactioncontroller.cpp:954]: Seems that this method is unused

File: src/notation/internal/notationactioncontroller.cpp [hunk 3]
```diff
@@ -892,6 +901,70 @@ void NotationActionController::addAnchoredNoteLine()
...
 }
 
+void NotationActionController::addStretch(qreal value)
+{
+    auto interaction = currentNotationInteraction();
+    if (!interaction) {
+        return;
+    }
+
+    if (!interaction->selection()->isRange()) {
```
Comment (RomanPudashkin) [inline src/notation/internal/notationactioncontroller.cpp:911]: Better to use currentNotationSelection()

File: src/notation/inotationinteraction.h [hunk 4]
```diff
@@ -117,6 +117,13 @@ class INotationInteraction
 
     virtual void addText(TextType type) = 0;
     virtual void addFiguredBass() = 0;
+
+    virtual void addStretch(qreal value) = 0;
+
+    virtual void resetStretch() = 0;
+    virtual void resetTextStyleOverrides() = 0;
```
Comment (RomanPudashkin) [inline src/notation/inotationinteraction.h:124]: Maybe better to unite those methods in one?

void resetToDefault(ResettableValueType type)

enum ResettableValueType {
Stretch,
BeamMode,
ShapesAndPosition,
TextStyleOverriders
}

File: src/appshell/view/appmenumodel.cpp (hunk 1 above)
Comment (Eism) [reply src/appshell/view/appmenumodel.cpp:260]: this is how it works now

File: src/notation/inotationinteraction.h (hunk 4 above)
Comment (Eism) [reply src/notation/inotationinteraction.h]: thanks, fixed

File: src/notation/internal/notationactioncontroller.cpp (hunk 3 above)
Comment (Eism) [reply src/notation/internal/notationactioncontroller.cpp]: thanks, fixed

File: src/notation/internal/notationactioncontroller.cpp (hunk 2 above)
Comment (Eism) [reply src/notation/internal/notationactioncontroller.cpp:970]: thanks, fixed

**Review by RomanPudashkin — APPROVED** [review]

**Pull request merged**